modal-js/src/sandbox.ts

```diff
@@ -56,6 +61,33 @@ export class Sandbox {
     );
   }
 
+  /**
+   * Open a file in the sandbox filesystem.
+   * @param path - Path to the file to open
+   * @param mode - File open mode (r, w, a, r+, w+, a+)
+   * @returns Promise that resolves to a SandboxFileHandle
+   */
+  async open(
+    path: string,
+    mode: SandboxFileMode = "r",
+  ): Promise<SandboxFileHandle> {
+    const taskId = await this.#getTaskId();
+    const resp = await runFilesystemExec({
+      fileOpenRequest: {
+        path,
+        mode,
+      },
+      taskId: taskId,
+    });
+    const fileDescriptor = resp.response.fileDescriptor;
+
+    if (!fileDescriptor) {
+      throw new Error(`Failed to open file ${path} with mode ${mode}`);
+    }
```

Comment:
See comment / question here about error handling on this method https://github.com/modal-labs/libmodal/pull/35#discussion_r2169322065